docs/source/logging/logging.md

```diff
@@ -1,54 +1,125 @@
 
-# Default framework-side logging configuration
+By default, Python only shows logging messages at level `WARNING` and above. Kedro's logging configuration specifies that `INFO` level messages from Kedro should also be emitted. This makes it easier to track the progress of your pipeline when you perform a kedro run.
 
-Kedro's [default logging configuration](https://github.com/kedro-org/kedro/blob/main/kedro/framework/project/default_logging.yml) defines a handler called `rich` that uses the [Rich logging handler](https://rich.readthedocs.io/en/stable/logging.html) to format messages. We also use the [Rich traceback handler](https://rich.readthedocs.io/en/stable/traceback.html) to render exceptions.
+## How to perform logging in your Kedro project
+To add logging to your own code (e.g. in a node):
 
-By default, Python only shows logging messages at level `WARNING` and above. Kedro's logging configuration specifies that `INFO` level messages from Kedro should also be emitted. This makes it easier to track the progress of your pipeline when you perform a `kedro run`.
+```python
+import logging
 
-## Project-side logging configuration
+log = logging.getLogger(__name__)
+log.warning("Issue warning")
+log.info("Send information")
+log.debug("Useful information for debugging")
+```
 
-In addition to the `rich` handler defined in Kedro's framework, the [project-side `conf/base/logging.yml`](https://github.com/kedro-org/kedro/blob/main/kedro/templates/project/%7B%7B%20cookiecutter.repo_name%20%7D%7D/conf/base/logging.yml) defines two further logging handlers:
-* `console`: show logs on standard output (typically your terminal screen) without any rich formatting
-* `info_file_handler`: write logs of level `INFO` and above to `info.log`
+```{note}
+The name of a logger corresponds to a key in the `loggers` section of the logging configuration file (e.g. `kedro`). See [Python's logging documentation](https://docs.python.org/3/library/logging.html#logger-objects) for more information.
+```
 
-The logging handlers that are actually used by default are `rich` and `info_file_handler`.
+You can take advantage of Rich's [console markup](https://rich.readthedocs.io/en/stable/markup.html) in your logging calls:
 
-The project-side logging configuration also ensures that [logs emitted from your project's logger](#perform-logging-in-your-project) should be shown if they are `INFO` level or above (as opposed to the Python default of `WARNING`).
+```python
+log.error("[bold red blink]Important error message![/]", extra={"markup": True})
+```
 
-We now give some common examples of how you might like to change your project's logging configuration.
+## How to customise Kedro logging
 
 ### Using `KEDRO_LOGGING_CONFIG` environment variable
```

Comment:
The hierarchy of this section feels wrong because there's no other way to customise logging now.

"How to customise Kedro logging" is basically the same as "Using KEDRO_LOGGING_CONFIG environment variable", and then "Show DEBUG level messages" is really just another example of a sort of customisation you want to make. All the other examples are currently given under "Advanced logging". So I think we should reorganise this a bit.

Reply:
How about we remove the sub-title "Using `KEDRO_LOGGING_CONFIG` environment variable" and just say that you need to use it for customisation. I'd like to keep the "How to customise Kedro logging" section heading but see what you're saying about the logic of the hierarchy. I've suggested the changes needed.

Reply:
I have incorporate this comment partially. I still think it is valuable to put the DEBUG level example first, since this is a common thing to do compare to other niche advance use cases. I've reorder things a bit so it is more consistent.